### DISCUSSION TITLE
Retrieving LngLat from feature (onTileLoad)

### DISCUSSION THREAD

**Discussion by 29942016** [visgl/deck.gl#5326] [Q&A] [answered]
Hi, just wondering if there's a way to retrieve the LngLat of a feature when using the MvtLayer? Currently when a tile is retrieved I check the MvtLayer's onTileLoad callback, find the feature I need and save it to state so I can pass it to another layer. 

Below is a sample feature retrieved. I believe its coordinates are already translated and irreversible?
![图片](https://user-images.githubusercontent.com/5634187/104397986-df961400-5588-11eb-8d19-9e47c325482b.png)

Basically I'm trying to retrieve any features of the text type, so i can render them separately in a text layer. Hopefully there's a simple solution for this, or another way I can go about it.

Heres the full callback for reference
```
  onTileLoad: (tile) => {
      tile.content.forEach(feature => {
          if (feature.properties.SubClasses.toLowerCase().endsWith("text")) {
              let textEnts;
              if (this.state.textEntities) {
                  textEnts = this.state.textEntities;
                  if (!textEnts.has(feature.properties.EntityHandle)) {
                      textEnts.set(feature.properties.EntityHandle, feature);
                  }
              } else {
                  textEnts = new Map();
                  textEnts.set(feature.properties.EntityHandle, feature);
              }

              this.setState({ textEntities: textEnts });
          }
      })
  },
```

Thanks!

Edit: I've noticed the bounding box is retrieved with the tile, is there a way to calculate exact a feature's lnglat position using a mixture of the tilebounds + feature x/y?

**Comment by Pessimistress** [comment, marked as answer]
This is already being worked on, will be available in 8.4.0.

**Reply by 29942016** [reply]
awesome, thanks :)
